### PR TITLE
Hotfix: Dependency version consolidation

### PR DESCRIPTION
**Pull request by kbwestfall**
Fixes some versioning issues.

### CODE REVIEW

**Comment by tbowers7** [comment]
A comment more than a review: what other dependency versions do we need to check?

For instance, may want to bump `scipy` to `>=1.9` to allow Python3.11  (https://docs.scipy.org/doc/scipy/dev/toolchain.html), but our `numpy` dependence should already grab `scipy>=1.8`.

File: environment.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: pypeit
 channels:
   - defaults
 dependencies:
-  - python>=3.9,<3.12
+  - python>=3.10,<3.13
```
Comment (VascoSch92) [inline environment.yml:5]: I don't know how much is rilevant but give a look here: https://dev.to/hugovk/the-python-3-1-problem-85g

Have you test the changes?

File: setup.cfg [hunk 2]
```diff
@@ -34,7 +34,7 @@ setup_requires = setuptools_scm
 include_package_data = True
 install_requires =
     numpy>=1.23
-    astropy>=4.3
+    astropy>=6.0
```
Comment (rcooke-ast) [inline setup.cfg:37]: I am fine with this change, of course, but I will note that `setup.cfg` changes here conflict with those made in PR #1755. I think one of these PRs need to revert the changes to this file.

File: setup.cfg [hunk 3]
```diff
@@ -34,7 +34,7 @@ setup_requires = setuptools_scm
...
     extension-helpers>=0.1
     packaging>=0.19
     scipy>=1.7
```
Comment (rcooke-ast) [inline setup.cfg:40]: Can we also update the ginga requirement to >=5.0.1, given the recent work by @ejeschke?

**Review by rcooke-ast — APPROVED** [review]
Approving but please consider the changes made in PR#1755 to see if this is a conflict. Thanks!

File: environment.yml (hunk 1 above)
Comment (badpandabear) [reply environment.yml:5]: I played around with this using yaml.safe_load and it parses "python>=3.10,<3.13" as a single string, so it should work. But it wouldn't be a bad idea to wrap them in quotes for safety anyway.

**Review by badpandabear — APPROVED** [review]
Looks good, but I second @rcooke-ast 's question about using ginga 5.0.1

**Comment by kbwestfall** [comment]
I mistakenly merged `develop` into this when I made the recent update, so I just redirected this PR.  After #1755 merges, we'll want to quickly move to a new released version anyway...

File: setup.cfg (hunk 3 above)
Comment (kbwestfall) [reply setup.cfg:40]: Done!

File: setup.cfg (hunk 2 above)
Comment (kbwestfall) [reply setup.cfg:37]: Thanks for pointing this out.  I'm working on consolidating this...

File: environment.yml (hunk 1 above)
Comment (kbwestfall) [reply environment.yml:5]: Thanks for pointing this out, @VascoSch92 , but, given @badpandabear 's comment, I think it's safe to leave it as is.

**Pull request merged**